src/wallet/wallet.cpp

```diff
@@ -2116,24 +2116,25 @@ void CWallet::CommitTransaction(CTransactionRef tx, mapValue_t mapValue, std::ve
         return true;
     });
 
+    // wtx can only be null if the db write failed.
+    if (!wtx) {
+        throw std::runtime_error(std::string(__func__) + ": Wallet db error, transaction commit failed");
```

Comment:
> Might be simpler and easier to maintain if this is raised directly at the source?

At least for now (with the current code architecture), as `AddToWallet` is called from `CommitTransaction`, `importPrunedFunds` and  `AddToWalletIfInvolvingMe` (and this last one is used in flows like chain sync and chain rescan), would keep it as is so we are able to throw different error descriptions on the method's caller side.
User and debugging wise, isn't the same to throw an error for a failure in the transaction commit flow than a failure storing a transaction when the node is syncing or rescanning the chain. They should be presented differently to the user and logs.

Nevertheless, I agree that it's a good point to take into account moving forward 👍🏼. We could improve the error handling returning a db failure status instead of the obscure `nullptr`.. (which originated other problematic -> #25272).

> Edit: Would it be helpful to indicate with which tx the db write failed?

I don't think that is needed on this specific process. `CommitTransaction` is trigger directly from a user action and the user does not know which is the id of the tx that the wallet is creating for him/her (he/she just pressed a button in the GUI or called to RPC commands like `sendtoaddress`, `send`, `sendall`, etc).
In other words, as a user, if I press the send button on the GUI (or call `sendtoaddress`), and the process fails, the tx id on the error message isn't helpful for me. It would actually be confusing.